### PR TITLE
Feat/babel support

### PR DESCRIPTION
**Pull request by tawashley**
## Add in babel support
Added option to the config file to enable babel transpilation. This option is turned `off` by default, with all source files being run through babel before concatenation when the option is turned `on`.

## Todos

- [x] Does this feature run on all supported platforms?
- [x] Is there documentation for this feature?
- [ ] Are there tests around this feature?

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015"]
```
Comment (furzeface) [inline .babelrc]: :octocat:

File: README.md [hunk 2]
```diff
@@ -34,8 +41,9 @@ This module provides the following gulp tasks
 * `gulp scripts:bundle` - Concatenates all JS files into one file. When the `--prod` is provided the concatenated file is also minified. Sourcemaps are generated by default when no `--prod` flag is provided.
 * `gulp scripts:lint` - Lints all JS files. Ignored if `--prod` is provided.
 * `gulp scripts:docs` - Generates JS docs, for when [JSDocs](http://usejsdoc.org/) are used in source file.
+* `gulp scripts::ensure-babel-symlink` - This is run with `scripts:bundle` and is only needed to run individually when debugging. This ensures the ES2015 Babel preset files are setup in the project.
```
Comment (furzeface) [inline README.md]: Should this be `gulp scripts:ensure-babel-symlink` @tawashley?

Comment (tawashley) [inline README.md]: @furzeface Yes it should, good eye

**Comment by bmds** [comment]
Looks like a good time for us to start moving to Eslint for this as well

**Comment by bmds** [comment]
Not with this Pull request obvs

**Comment by tawashley** [comment]
@bmds Yeah, I want to look at compile to multiple files and then revamping the tests to use the new gulp runner next. ESLint makes sense after - being a breaking / major version

**Comment by furzeface** [comment]
I'm happy with this PR, second opinion before merge, @bmds?

**Comment by bmds** [comment]
Yes i'm happy with this too.
Making it optional allows a minor version bump.
We'll need to do a major release when it becomes the default

**Pull request merged**